### PR TITLE
Lets you cut the plastic tip off of toy guns to make them look real

### PR DESCRIPTION
**Pull request by dieamond13**
## About The Pull Request
Using wirecutters, you can snip the plastic barrel tip off of donksoft toy guns to make them look identical to real guns. If you take a closer look at the "real" gun it'll look a little cheap though..
## Why It's Good For The Game
I think it's pretty funny and I don't know why it hasn't been implemented. In real life people cut off the tips of airsoft / other fake guns to make them look more real too so I thought it would be cool to have that in game too.
## Changelog
:cl:
add: Lets you cut the plastic tips off of toy guns to make them look real.
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
people throw the word soul around a lot but they never quite land it

this is soul

File: code/modules/projectiles/guns/ballistic/toy.dm [hunk 1]
```diff
@@ -15,6 +15,18 @@
 	gun_flags = TOY_FIREARM_OVERLAY | NOT_A_REAL_GUN
 	casing_ejector = FALSE
 
+/obj/item/gun/ballistic/automatic/toy/wirecutter_act(mob/living/user, obj/item/I)
```
Comment (nikothedude) [inline code/modules/projectiles/guns/ballistic/toy.dm:18]: i really suggest moving all this down to /automatic/toy, and just adding a variable/flag that says "can we have our tip removed/do we have a tip", and if its true, do all the stuff, but replace the name/desc setting to be a kinda

```
name = get_untipped_name() //where this proc is on /obj/item/gun/ballistic/automatic/toy
// same for desc
```

Comment (nikothedude) [reply code/modules/projectiles/guns/ballistic/toy.dm:18]: wait nevermind theres no basetype for toy guns - move it down to item/gun but still check the aformentioned vars/flags

Comment (Fikou) [reply code/modules/projectiles/guns/ballistic/toy.dm:18]: i hate you

File: code/modules/projectiles/guns/ballistic/toy.dm [hunk 2]
```diff
@@ -15,6 +15,18 @@
...
+	to_chat(user, span_notice("You start cutting off the plastic tip..."))
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic/toy.dm:19]:
```suggestion
	balloon_alert(user, "cutting off tip...")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Too much copy-paste, make this a component.

File: code/modules/projectiles/guns/ballistic/toy.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/projectiles/guns/ballistic/toy.dm:18]: Too niche to be on the base subtype, should be component instead

Comment (dieamond13) [reply code/modules/projectiles/guns/ballistic/toy.dm:18]: > 

how do I make it so each gun knows which name/desc to pick if all of them are get_untipped_name()?

Comment (Fikou) [reply code/modules/projectiles/guns/ballistic/toy.dm:18]: make the component pass through the path of the "non toy" version and it copies vars from that path

**Comment by DATA-xPUNGED** [comment]
Space Law should be updated to make this illegal if it gets in

**Comment by Shadowflame909** [comment]
Need the inverse to be possible as well with a yellow crayon and washing machine. Very cool!

**Comment by nikothedude** [comment]
DONT LET THE FLAME DIE OUT

**Comment by dieamond13** [comment]
this is on hold for a little bit until I figure out how to do some coder things

**Pull request closed without merging**